### PR TITLE
Fix server data folder not created & test races

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Fixes `temporal server start-dev` not creating the folder for Data/DB

## Why?

The issue with `temporal server start-dev` not creating the data folder was exposed by merging https://github.com/temporalio/tctl-kit/pull/35 : the `env` feature was creating the folder so the tests were succeeding

## Checklist

1. Closes 

2. How was this tested:

Unit tests

3. Any docs updates needed?

### CODE REVIEW

**Review by robholland — APPROVED** [review]

File: app/app_test.go [hunk 1]
```diff
@@ -320,68 +319,77 @@ func assertServerHealth(t *testing.T, ctx context.Context, opts client.Options)
...
-func TestCreateDataDirectory(t *testing.T) {
+func TestCreateDataDirectory_MissingDirectory(t *testing.T) {
+	temporalCLI := app.BuildApp("")
+	// Don't call os.Exit
+	temporalCLI.ExitErrHandler = func(_ *cli.Context, _ error) {}
+
+	portProvider := sconfig.NewPortProvider()
+	var (
+		port = portProvider.MustGetFreePort()
+	)
```
Comment (cretz) [inline app/app_test.go:330]:
```suggestion
	port := portProvider.MustGetFreePort()
```

(not really reviewing the PR, just saw this as I passed by)

**Comment by feedmeapples** [comment]
https://github.com/temporalio/temporal/issues/3708 seems to be causing the nil pointer exception with SQLite driver

File: app/app_test.go [hunk 2]
```diff
@@ -320,68 +319,75 @@ func assertServerHealth(t *testing.T, ctx context.Context, opts client.Options)
...
+func TestCreateDataDirectory_ExistingDirectory(t *testing.T) {
 	temporalCLI := app.BuildApp("")
 	// Don't call os.Exit
 	temporalCLI.ExitErrHandler = func(_ *cli.Context, _ error) {}
 
 	portProvider := sconfig.NewPortProvider()
 	var (
-		port1 = portProvider.MustGetFreePort()
-		port2 = portProvider.MustGetFreePort()
+		port = portProvider.MustGetFreePort()
```
Comment (robholland) [inline app/app_test.go:353]: Same as above, should just be `port :=`

**Review by robholland — APPROVED** [review]

**Comment by feedmeapples** [comment]
flakiness comes from https://github.com/temporalio/temporal/issues/3708
merging

**Pull request merged**